machine.go

```diff
@@ -530,6 +533,28 @@ func (m *Machine) startInstance(ctx context.Context) error {
 	return err
 }
 
+// SetMetadata sets the machine's metadata for MDDS
+func (m *Machine) SetMetadata(ctx context.Context, metadata interface{}) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
```

Comment:
Why do we need `cancel` here?

Reply:
TBH I copied the pattern from `startInstance`. I suspect it's there to cancel out any kept alive connections, but I don't really know.
I'll be happy to remove this from here and `startInstance` if you think that's the correct thing to do.

Reply:
I don't think it's needed. Please do so.